clientapi/routing/login.go

```diff
@@ -40,28 +41,26 @@ type flow struct {
 	Type string `json:"type"`
 }
 
-func passwordLogin() flows {
-	f := flows{}
-	s := flow{
-		Type: "m.login.password",
-	}
-	f.Flows = append(f.Flows, s)
-	return f
-}
-
 // Login implements GET and POST /login
 func Login(
 	req *http.Request, userAPI userapi.ClientUserAPI,
 	cfg *config.ClientAPI,
 ) util.JSONResponse {
 	if req.Method == http.MethodGet {
-		// TODO: support other forms of login other than password, depending on config options
+		var loginFlows []flow
+		loginFlows = []flow{{Type: authtypes.LoginTypePassword}}
+		if len(cfg.Derived.ApplicationServices) == 0 {
+			loginFlows = append(loginFlows, flow{Type: authtypes.LoginTypeApplicationService})
+		}
```

Comment:
```suggestion
		loginFlows := []flow{{Type: authtypes.LoginTypePassword}}
		if len(cfg.Derived.ApplicationServices) > 0 {
			loginFlows = append(loginFlows, flow{Type: authtypes.LoginTypeApplicationService})
		}
```
You don't need to create a variable first, you can directly use `:=` to create and assign a value to a variable in Go. :)
Also, we want to add that loginFlow if there are > 0 defined services.

Reply:
Ahahaha, sorry, my bad, i copied the len(..) stuff from somewhere else in the source (instead of from you) and forgot to change the == to >, my bad 😓 
Also, thanks for the info regarding `:=`, learning a bunch here 👍